### DISCUSSION TITLE
Deleting replications results in a timeout

### DISCUSSION THREAD

**Discussion by niklasweimann** [goharbor/harbor#18969] [Q&A]
Hey 👋 
i want to remove several replications which we used to push images to docker hub for about an year. So the Replication got executed quite often. When i try to delete the replication i always got a timeout for the request so i updated the ingress with in Timeout of 3600s which is fine, but the delete request still never succeeded.

Next i checked the code and found out that the request first looks up all executions for this policy than all tasks for the execution and fails as soon as any of the executions is not in a final state. But i checked this condition in the database and all executions where in a final state. So i counted the executions and there were about 50k tasks that should be deleted.
``` SQL
select count(*) from execution e left join task t on e.id = t.execution_id where e.vendor_id = 14 and e.vendor_type = 'REPLICATION';
```

I guess that the deletion process is taking too much time so that the request never succeeds. Is there a way to improve the speed? I would prefer not to delete the entries from the database by hand.

**Comment by chlins** [comment]
Track this by issue https://github.com/goharbor/harbor/issues/18988.
